internal/utils/url.go

```diff
@@ -89,3 +91,13 @@ func GetURLWithScheme(rawURL string) (string, error) {
 
 	return rawURL, nil
 }
+
+func GenerateTenantURL(baseURL string, tenantID string) (string, error) {
+	protocol, urlName, validURL := strings.Cut(baseURL, urlSeparator)
+	if !validURL || protocol == "" || urlName == "" {
+		return "", fmt.Errorf("invalid base URL: %s", baseURL)
+	}
+
+	tntURL := protocol + urlSeparator + tenantID + "." + urlName
+	return tntURL, nil
+}
```

Comment:
I think this would be more robust if we rely on `url.Parse()` for parsing and validating the URL. There are a lot of edge cases we'd have to consider otherwise. This should pass the unit tests. 

```go
func GenerateTenantURL(baseURL string, tenantID string) (string, error) {
	if tenantID == "" {
		return "", fmt.Errorf("tenantID is empty")
	}

	parsedURL, err := url.Parse(baseURL)
	if err != nil {
		return "", fmt.Errorf("invalid base URL %s: %w", baseURL, err)
	}

	hostParts := strings.SplitN(parsedURL.Hostname(), ".", 2)
	if len(hostParts) < 2 {
		return "", fmt.Errorf("base URL must have at least two domain parts %s", baseURL)
	}

	newHostname := fmt.Sprintf("%s.%s.%s", tenantID, hostParts[0], hostParts[1])
	if parsedURL.Port() != "" {
		parsedURL.Host = newHostname + ":" + parsedURL.Port()
	} else {
		parsedURL.Host = newHostname
	}

	return parsedURL.String(), nil
}
```

Reply:
thanks, these are great suggestions for checking the URL's integrity [8670fc1](https://github.com/stellar/stellar-disbursement-platform-backend/pull/281/commits/8670fc15de3949ca544acde78864904bec6d22e7)